src/tools/kdb/complete.cpp

```diff
@@ -102,6 +106,17 @@ void CompleteCommand::addMountpoints (KeySet & ks, const Key root)
 	printWarnings (cerr, mountpointPath);
 }
 
+void CompleteCommand::addNamespaces (map<Key, pair<int, int>> & hierarchy)
+{
+	// Currently assumed to be fixed, and they are on level 0
+	const string namespaces[] = { "spec/", "proc/", "dir/", "user/", "system/" };
```

Comment:
Yes, it will return a hard-coded list of namespaces. But the function should return a warning if we choose to extend the namespaces. It is explained in the docu of `keyGetNamespace`, simply do it like in the loop example there. https://doc.libelektra.org/api/latest/html/group__keyname.html#gafc3ca03ed10f87eb59bdc02cf2a0de8d

But you are right, it might be overkill to add such trivial code in the binding or libtools.